src/dao/apiLogs.dao.ts

```diff
@@ -10,5 +10,47 @@ export class APILogDAO extends BaseDAO {
     this.model = APILogModel;
   }
 
+  async getLatestInvocations(query: any, limit: number, offset: number) {
+    const invocations = await this.model
+      .find(query)
+      .sort({ timestamp: -1 })
+      .skip(offset)
+      .limit(limit);
 
-}
\ No newline at end of file
+    const totalCount = await this.model.countDocuments(query);
+    return { invocations, totalCount };
+  }
+
+  async getAverageResponseTimeByFilters(query: any) {
```

Comment:
the avaerage response time should be w.r.t to endpoints
we can remove remaining filters